tests/storage-metadata

```diff
@@ -0,0 +1,342 @@
+    lxc config device add c1 vhost-vsock unix-char source=/dev/vhost-vsock
+    lxc config device add c1 vsock unix-char source=/dev/vsock
+
+    # workaround for LP: #2104066
+    lxc exec c1 -- mkdir -p /etc/systemd/system/snapd.service.d
+    lxc file push - c1/etc/systemd/system/snapd.service.d/override.conf << EOF
+# Workaround for https://bugs.launchpad.net/snapd/+bug/2104066
+[Service]
+Environment=SNAPD_STANDBY_WAIT=1m
+EOF
+    lxc exec c1 -- systemctl daemon-reload
+    lxc exec c1 -- systemctl restart snapd.service
+
+    # shellcheck disable=SC3044 # Ignore "declare is undefined" shellcheck error.
+    lxc exec c1 -- sh -c "$(declare -f waitSnapdSeed); waitSnapdSeed"
+    lxc exec c1 -- snap install lxd --channel="${versStr}"
+    lxc exec c1 -- lxd init --auto --storage-backend=dir --network-address="[::]" --network-port=8443
+
+    echo "==> Add nested LXD as remote"
+    token="$(lxc exec c1 -- lxc config trust add --name host --quiet)"
+    lxc remote add c1 "${token}"
+
+    tmpdir="$(mktemp -d)"
+
+    # Check if instance backups from ${versStr} can be imported into present version of LXD.
+    for type in container vm; do
+        extra_args=""
+        if [ "${type}" = "vm" ]; then
+            extra_args="--vm"
+        fi
+
+        echo "==> Create a ${type} and snapshot it"
+        lxc init c1:i1 -d root,size=1MiB --empty ${extra_args} # Don't quote to omit empty extra args.
+        lxc snapshot c1:i1
+
+        echo "==> Export using the default backup format implicitly by not specifying a version"
+        # 5.0/5.21 always use the old format.
+        # > 5.21 uses the new format.
+        lxc export c1:i1 "${tmpdir}/i1.tar.gz"
+        tar -xzf "${tmpdir}/i1.tar.gz" -C "${tmpdir}" backup/index.yaml
+
+        echo "==> Check the backup index file"
+        cat "${tmpdir}/backup/index.yaml"
+
+        if [ "${vers}" = "5.21" ] || [ "${vers}" = "5.0" ]; then
+            check_format_version1 "${tmpdir}/backup/index.yaml" instance
+
+            # The old format should be allowed for import in any case.
+            echo "==> Check the ${type} can be imported on LXD ${LXD_SNAP_CHANNEL}"
+            lxc import "${tmpdir}/i1.tar.gz" i1
+            [ "$(lxc query "/1.0/instances/i1?recursion=1" | jq '.snapshots | length')" = "1" ]
+
+            lxc delete -f i1
+        else
+            check_format_version2 "${tmpdir}/backup/index.yaml" instance
+
+            # Only LXD >= 5.21 allows importing the new format.
+            if hasNeededAPIExtension backup_metadata_version; then
+                echo "==> Check the ${type} can be imported on LXD ${LXD_SNAP_CHANNEL}"
+                lxc import "${tmpdir}/i1.tar.gz" i1
+                [ "$(lxc query "/1.0/instances/i1?recursion=1" | jq '.snapshots | length')" = "1" ]
+
+                lxc delete -f i1
+            fi
+        fi
+
+        rm -rf "${tmpdir}/backup"
+        rm "${tmpdir}/i1.tar.gz"
+
+        # Only LXD >= 5.21 allows setting the export version.
```

Comment:
wont this comment be out of date once we backport support for v2 format to 5.21?